### ISSUE TITLE
clazy

### ISSUE THREAD

**Issue by atsju** [githubdoe/DFTFringe#67]
I just managed to run clazy with exclusion pattern to get only warnings from DFTF code.

Total 8697 warnings
But 8512 of them are [connect-by-name](https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-by-name.md) and I suppose we must ignore them because nobody plans on updating this

This means we have 185 more meaningfull warns. Not too bad.

clang also generated some `[-Woverloaded-virtual]` and `[-Wabsolute-value]`. I will probably fix the `[-Wabsolute-value]`

**Comment by atsju** [comment]
84 [-Wclazy-overloaded-signal]
34 [-Wclazy-unused-non-trivial-variable]
6 [-Wclazy-container-anti-pattern]
35 [-Wclazy-fully-qualified-moc-types]
11 [-Wclazy-detaching-temporary]
7 [-Wclazy-incorrect-emit]
1 [-Wclazy-qdatetime-utc]
1 [-Wclazy-qstring-insensitive-allocation]
2 [-Wclazy-qmap-with-pointer-key]
4 [-Wclazy-qstring-arg]

**Comment by atsju** [comment]
@gr5 @githubdoe 
Just an information comment (I do not plan any change unless you want me to do it). And no, I do not work on fixing clazy warning, but I understood this while fixing other problems.

As I told you that clazy was warning on [connect-by-name](https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-by-name.md) meaning ui signals were connected using `on_buttonName_clicked()`.
At the time of the discussion, we didn't know what was correct method to use instead and I promised I keep you updated if I found out. Now I found out and its a bit unclear but in QT documentation [here](https://doc.qt.io/qt-6/designer-using-a-ui-file.html#automatic-connections)
And unsurprisingly, what we do (`on_buttonName_clicked()`) is officially not recommended in QT6 documentation:
> Note: When renaming widgets in the form, the slot names need to be adapted accordingly, which can become a maintenance problem. **For this reason, we recommend against using this in new code.**

It took me some time to understand the documentation so here is the summary:
Instead of  `on_buttonName_clicked()` auto-connect meathod, one should create slot `my_buttonName_clicked()` and connect it in constructor with `connect(buttonName, &QAbstractButton::clicked, this, &theDialog::my_buttonName_clicked);` to use manual connection method.

It's as simple as that and the recommandation root cause is straightforward. Auto-connect is runtime and you won't know until you test it and it fails. Manual connect is compile time checked.
All fears about "don't touch ui files or it might break signals" are solved.

In my example I changed function name only from `on_buttonName_clicked()` to `my_buttonName_clicked()` because I strongly agree this naming convention is good and simple.
Again I don't plan on changing anything but over about 8000 signals, there is probably at least 1 that doesn't work as expected.
